.github/workflows/package-ubuntu.yml

```diff
@@ -0,0 +1,207 @@
+      - name: setup build environment
+        run: |
+          export CXXFLAGS="$CXXFLAGS"
+          if [[ "${{matrix.clang}}" != "11" && "${{matrix.clang}}" -lt "16" ]]; then
+            export OMP_CXX_FLAGS="$CXXFLAGS -fexperimental-new-pass-manager"
+            export CC=clang-${{matrix.clang}}
+            export CXX=clang++-${{matrix.clang}}
+          fi
+          echo "CC=${CC}" >> $GITHUB_ENV
+          echo "CXX=${CXX}" >> $GITHUB_ENV
+          echo "OMP_CXX_FLAGS=${OMP_CXX_FLAGS}" >> $GITHUB_ENV
+      - name: install CUDA
+        run : |
+          mkdir -p /opt/AdaptiveCpp/cuda
+          wget -q -O cuda.sh http://developer.download.nvidia.com/compute/cuda/11.0.2/local_installers/cuda_11.0.2_450.51.05_linux.run
+          sudo sh cuda.sh --override --silent --toolkit --no-man-page --no-drm --no-opengl-libs --installpath=/opt/AdaptiveCpp/cuda || true
+          echo "CUDA Version ${{matrix.cuda}}" | sudo tee /opt/AdaptiveCpp/cuda/version.txt
+          rm cuda.sh
+          sudo apt install -y libcudart$(echo ${{matrix.cuda}} | sed 's/..$//')
+      - name: install ROCm
+        run: |
+          [[ ${{matrix.os}} == ubuntu-22.04 ]] && CODENAME=jammy
+          sudo apt install libnuma-dev cmake unzip
+          wget -q -O - https://repo.radeon.com/rocm/rocm.gpg.key | sudo apt-key add -
+          echo "deb [arch=amd64] https://repo.radeon.com/rocm/apt/${{matrix.rocm}} $CODENAME main" | sudo tee /etc/apt/sources.list.d/rocm.list
+          printf 'Package: *\nPin: release o=repo.radeon.com\nPin-Priority: 600' | sudo tee /etc/apt/preferences.d/rocm-pin-600
+          sudo apt update
+          sudo apt install rocm-dev
+      - name: install OpenCL development files and loader
+        run: |
+          wget -O- https://apt.repos.intel.com/intel-gpg-keys/GPG-PUB-KEY-INTEL-SW-PRODUCTS.PUB | gpg --dearmor | sudo tee /usr/share/keyrings/oneapi-archive-keyring.gpg > /dev/null
+          echo "deb [signed-by=/usr/share/keyrings/oneapi-archive-keyring.gpg] https://apt.repos.intel.com/oneapi all main" | sudo tee /etc/apt/sources.list.d/oneAPI.list
+          sudo apt update
+          sudo apt install -y intel-oneapi-runtime-opencl-2024 intel-oneapi-runtime-compilers-2024 ocl-icd-libopencl1 ocl-icd-opencl-dev
+      - name: install Level Zero
+        run : |
+          wget https://github.com/oneapi-src/level-zero/releases/download/v1.2.3/level-zero-devel_1.2.3+u18.04_amd64.deb
+          wget https://github.com/oneapi-src/level-zero/releases/download/v1.2.3/level-zero_1.2.3+u18.04_amd64.deb
+          sudo dpkg -i ./level-zero*
+      - name: install packaging helpers
+        run: |
+          sudo  apt install build-essential
+          sudo  apt install devscripts
+          sudo  apt install debhelper
+      - name: package AdaptiveCpp
+        run: |
+          cd ..
+          cp -R AdaptiveCpp acpp-24.02.0
+          tar -cf acpp_24.02.0.orig.tar.gz acpp-24.02.0/
```

Comment:
Does it actually need to be versioned here? Ultimately we'd like to have new packages for every push anyway, and it would be easier to maintain if we didn't have to always update version numbers.

Reply:
yes, the tar file needs to carry this name-version scheme. Part of the initialization process is to create an upstream tarball <name>_<version>-<package_revision>.tar.xz